systemtest/src/main/java/hivemall/systemtest/runner/SystemTestRunner.java

```diff
@@ -0,0 +1,317 @@
+/*
+ * Hivemall: Hive scalable Machine Learning Library
+ *
+ * Copyright (C) 2016 Makoto YUI
+ * Copyright (C) 2013-2015 National Institute of Advanced Industrial Science and Technology (AIST)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package hivemall.systemtest.runner;
+
+import com.google.common.io.Resources;
+import hivemall.systemtest.model.CreateTableHQ;
+import hivemall.systemtest.model.DropTableHQ;
+import hivemall.systemtest.model.HQ;
+import hivemall.systemtest.model.InsertHQ;
+import hivemall.systemtest.model.RawHQ;
+import hivemall.systemtest.model.StrictHQ;
+import hivemall.systemtest.model.TableHQ;
+import hivemall.systemtest.model.TableListHQ;
+import hivemall.systemtest.model.UploadFileAsNewTableHQ;
+import hivemall.systemtest.model.UploadFileHQ;
+import hivemall.systemtest.model.UploadFileToExistingHQ;
+import hivemall.systemtest.utils.IO;
+import org.hamcrest.Matchers;
+import org.junit.Assert;
+import org.junit.rules.ExternalResource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+
+public abstract class SystemTestRunner extends ExternalResource {
+    static final Logger logger = LoggerFactory.getLogger(SystemTestRunner.class);
+    List<StrictHQ> classInitHqs = new ArrayList<StrictHQ>();
+    Set<String> immutableTables = new HashSet<String>();
+    final String dbName;
+    final Properties props;
+
+
+    SystemTestRunner(SystemTestCommonInfo ci, String propertiesFile) {
+        this.dbName = formatDBName(ci.dbName);
+
+        final String path = "hivemall/" + propertiesFile;
+        try {
+            InputStream is = null;
+            try {
+                props = new Properties();
+                is = new FileInputStream(Resources.getResource(path).getPath());
+                props.load(is);
+            } finally {
+                if (is != null)
+                    is.close();
+            }
+        } catch (Exception ex) {
+            throw new RuntimeException("Failed to load properties from " + path + ". "
```

Comment:
better to avoid RuntimeException and use subclasses of RuntimeException for more appropriate errors.